### PR TITLE
Fix DetailsList onRenderColumnHeaderTooltip not rendering >= 6.18.0

### LINKED ISSUE #5703: DetailsList onRenderColumnHeaderTooltip does not work beyond version 6.18.0
### Bug Report 
 
- __Package version(s)__: 6.18.1
- __Browser and OS versions__: All browsers
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: Normal
  
#### Describe the issue:
 
We have used the DetailsList and used the **onRenderDetailsHeader** (which internally calls **onRenderColumnHeaderTooltip**) to customize the header to add a tooltip. It worked when I was using officefabric version **6.1.2** but when upgraded to version beyond 6.18.1 the implementation is broken and I am not able to see the tooltip.
 
#### Actual behavior:
 
Not able to customize the header using onRenderDetailsHeader 
 
#### Expected behavior:
 
Should be able to customize the tooltip in the header using onRenderDetailsHeader prop

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #5703
- [x] Include a change request file using `$ npm run change`

### Description of changes

As noted in #5703, `onRenderColumnHeaderTooltip` has not worked in OUIFR versions `>= 6.18.0`.  This PR fixes that regression by allowing for developers to supply a `onRenderColumnHeaderTooltip` via `onRenderDetailsHeader`. By default, we do not render column header tooltips _by design_.

If the default props are used when providing `renderAs` overrides, then the `IColumn`'s `ariaLabel` will be used as the tooltip's contents on `mouseenter`.

### Focus areas to test

- By default `DetailsList` should not render tooltips for column headers (existing behavior).
- If `onRenderDetailsHeader` is provided, which in-turn passes a render function for the `onRenderColumnHeaderTooltip` prop, then column header tooltips should render on `mouseenter`.

#### Screengrabs

**Before**

dl_tooltip_basic_before

**After**

dl_tooltip_basic_after

#### Testing Locally

Here is a diff that can be applied to the "Basic DetailsList" example which exhibits the above behavior. I've also added unit tests to ensure that the render functions are invoked if provided.

```diff
diff --git a/packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Basic.Example.tsx b/packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Basic.Example.tsx
index a4f13c48d..0242d530f 100644
--- a/packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Basic.Example.tsx
+++ b/packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Basic.Example.tsx
@@ -3,7 +3,10 @@ import { TextField } from 'office-ui-fabric-react/lib/TextField';
 import { DetailsList, DetailsListLayoutMode, Selection, IColumn, IDetailsList } from 'office-ui-fabric-react/lib/DetailsList';
 import { MarqueeSelection } from 'office-ui-fabric-react/lib/MarqueeSelection';
 import { Checkbox } from 'office-ui-fabric-react/lib/Checkbox';
-import { createRef } from 'office-ui-fabric-react/lib/Utilities';
+import { createRef, IRenderFunction } from 'office-ui-fabric-react/lib/Utilities';
+import { IDetailsHeaderProps } from '../DetailsHeader.types';
+import { DetailsHeader } from '../DetailsHeader';
+import { ITooltipHostProps, TooltipHost } from '../../Tooltip';
 
 const _items: any[] = [];
 
@@ -84,6 +87,7 @@ export class DetailsListBasicExample extends React.Component<
             items={items}
             columns={_columns}
             setKey="set"
+            onRenderDetailsHeader={this._onRenderDetailsHeader}
             layoutMode={DetailsListLayoutMode.fixedColumns}
             selection={this._selection}
             selectionPreservedOnEmptyClick={true}
@@ -103,6 +107,17 @@ export class DetailsListBasicExample extends React.Component<
     }
   }
 
+  private _onRenderDetailsHeader(props: IDetailsHeaderProps, defaultRender?: IRenderFunction<IDetailsHeaderProps>): JSX.Element | null {
+    return (
+      <DetailsHeader
+        {...props}
+        onRenderColumnHeaderTooltip={(props: ITooltipHostProps, defaultRender?: IRenderFunction<ITooltipHostProps>) => (
+          <TooltipHost {...props} />
+        )}
+      />
+    );
+  }
+
   private _getSelectionDetails(): string {
     const selectionCount = this._selection.getSelectedCount();
 ```

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
Looks like some tests that were written after this behavior regressed are failing. I'll post an iteration fixing them and explain the changes.

File: packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap [hunk 1]
```diff
@@ -1474,31 +1510,8 @@ exports[`DetailsHeader renders accessible labels 1`] = `
...
-          padding-bottom: 0px;
-          padding-left: 0px;
-          padding-right: 0px;
-          padding-top: 0px;
-          position: absolute;
-          width: 1px;
-        }
-    id="header4-b-tooltip"
-  >
-    Click to sort.
```
Comment (ThomasMichon) [inline packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap]: I don't see this text emitted anymore. These strings still need to remain as the content of the element pointed to by `aria-describedby`.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsColumn.base.tsx [hunk 2]
```diff
@@ -179,7 +187,7 @@ export class DetailsColumnBase extends BaseComponent<IDetailsColumnProps> {
     tooltipHostProps: ITooltipHostProps,
     defaultRender?: IRenderFunction<ITooltipHostProps>
   ): JSX.Element => {
-    return <span className={tooltipHostProps.hostClassName}>{tooltipHostProps.children}</span>;
+    return <TooltipHost {...tooltipHostProps} />;
```
[the remark this answers is not in the document]
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsColumn.base.tsx]: I've taken another approach given @ThomasMichon's context above, updating this branch forcefully. Thank you Thomas for raising awareness 👍.

File: packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap (hunk 1 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap]: No longer applicable, thank you for the heads-up though.

**Comment by KevinTCoughlin** [comment]
>This PR may be stamping over critical accessibility fixes involving the column header Aria labeling, especially with respect to sort status and button descriptions.

@ThomasMichon, may I request another review. I believe I've addressed the issues you raised regarding default behavior and accessibility regressions. Thank you for bringing those to my attention 👍.

**Pull request merged**